gadgets/trace_signal/test/unit/trace_signal_test.go

```diff
@@ -0,0 +1,124 @@
+// GADGET_TAG=trace-signal-unit-tests IG_VERIFY_IMAGE=false go test -v -exec 'sudo -E' ./gadgets/trace_signal/test/unit/…
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+)
+
+type traceSignalEvent struct {
+	Proc      utils.Process `json:"proc"`
+	Signal    string        `json:"sig"`
+	SignalRaw int           `json:"sig_raw"`
+	Error     string        `json:"error"`
+}
+
+type testDef struct {
+	name   string
+	signal syscall.Signal
+}
+
+func TestTraceSignalGadget(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+	testCases := []testDef{
+		{
+			name:   "kill",
+			signal: syscall.SIGKILL,
+		},
+		{
+			name:   "sigterm",
+			signal: syscall.SIGTERM,
+		},
+		{
+			name:   "sigint",
+			signal: syscall.SIGINT,
+		},
+		{
+			name:   "sigusr1",
+			signal: syscall.SIGUSR1,
+		},
+		{
+			name:   "sigusr2",
+			signal: syscall.SIGUSR2,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			runner := utilstest.NewRunnerWithTest(t, nil)
+
+			onGadgetRun := func(gadgetCtx operators.GadgetContext) error {
+				utilstest.RunWithRunner(t, runner, func() error {
+					p, err := os.StartProcess("/usr/bin/sleep", []string{"/usr/bin/sleep", "2"}, &os.ProcAttr{})
```

Comment:
It appears we either need to wait for the process using `.Wait()` or detach from it using `.Release()` in order to clean up the resources used for it: https://pkg.go.dev/os#Process.Release

I missed it for the `trace_exec` unit test too. Could you create a small PR fix for that unit test?